dockstore-webservice/src/main/java/io/dockstore/webservice/core/LimitedSourceFileBuilder.java

```diff
@@ -93,25 +94,34 @@ public SourceFile build() {
 
         private static void setContentWithLimits(SourceFile file, String content, String path) {
             String limitedContent = content;
-            if (content != null) {
+            SourceFile.FormEnum limitedForm = SourceFile.FormEnum.COMPLETE;
+            // Limit certain types of content.
+            if (content == null) {
+                limitedContent = "Dockstore could not retrieve this file";
+                limitedForm = SourceFile.FormEnum.ERROR;
+            } else {
                 byte[] bytes = content.getBytes(StandardCharsets.UTF_8);
+                if (Bytes.indexOf(bytes, Byte.decode("0x00")) != -1) {
+                    // Postgres cannot store strings that contain "NUL" characters.
+                    // Thus, Dockstore cannot currently store binary files.
+                    // https://www.postgresql.org/docs/current/datatype-character.html#DATATYPE-CHARACTER
+                    // https://www.ascii-code.com/character/%E2%90%80
+                    limitedContent = "Dockstore does not store binary files";
+                    limitedForm = SourceFile.FormEnum.ERROR;
+                }
                 long maximumSize = computeMaximumSize(path);
-                // A large file is probably up to no good.
                 if (bytes.length > maximumSize) {
+                    // A large file is probably up to no good.
                     limitedContent = String.format("Dockstore does not store files of this type over %.1fMB in size", maximumSize / (double) BYTES_PER_MEGABYTE);
-                }
-                // Postgresql cannot store strings that contain "NUL" characters.
-                // https://www.postgresql.org/docs/current/datatype-character.html#DATATYPE-CHARACTER
-                // https://www.ascii-code.com/character/%E2%90%80
-                if (Bytes.indexOf(bytes, Byte.decode("0x00")) != -1) {
-                    limitedContent = "Dockstore does not store binary files";
+                    limitedForm = SourceFile.FormEnum.ERROR;
```

Comment:
Borderline (see previous comment), but seems more okay

Reply:
To elaborate, I'd almost imagine that an error should really be something unexpected like a 500 on GitHub's side and we'd have states for large files or binary files rather than merging them into errors in-case those two states are useful someday.
i.e. maybe we'd link to binaries or large files on github?

Reply:
I guess this is similar to your description 

> Later, we could add other values to the "form" enum. For example, if we wanted to try to handle big files by storing some portion of the content (instead of totally jettisoning it), we could have values like:

Reply:
> we'd have states for large files or binary files 

IMHO, the following two things are different concepts:
* whether a file is binary (or not).
* how we choose to represent the file's content on our end.

I'd recommend they are stored in separate fields (one for `isBinary`, another that describes the representation we've chosen for the content (which could be "none").

Reply:
Makes sense, I think as long as they're not merged into the error state, being even more descriptive is totally cool